pymongo/uri_parser.py

```diff
@@ -141,7 +149,18 @@ def parse_userinfo(userinfo):
     # No password is expected with GSSAPI authentication.
     if not user:
         raise InvalidURI("The empty string is not valid username.")
-    return unquote_plus(user), unquote_plus(passwd)
+
+    # If we cannot round-trip a value, it is not properly URI encoded.
```

Comment:
This was done to ensure that we raise an exception when getting URI strings like the one in the test entitled "Username with password containing an unescaped percent sign". The URI is `"mongodb://alice%foo:bar@127.0.0.1"`.